torch/csrc/autograd/engine.cpp

```diff
@@ -1123,6 +1142,7 @@ void GraphTask::init_to_execute(Node& graph_root, const edge_list& outputs, bool
   std::vector<Frame> stack;
   std::unordered_set<Node*> seen;
   for (const auto & input : graph_root.next_edges()) {
+    if (!input.function.get()) continue;
```

Comment:
When can this actually happen? When you do `autograd.grad(a, a)`? If so can you add a comment here?

Reply:
Happens when you multiply a Tensor with a scalar for example `autograd.grad(a*2, a)` - I expected it to mul_scalar, but it actually calls mul_Tensor. I suspect that mul_Tensor when calls collect_edges, it returns a default constructed edge when because it receives an invalid Tensor.

Reply:
The python binding is doing this AFAIK
But the same would happen if you do the following no?
```python
a = torch.rand(2, requires_grad=True)
b = torch.rand(2)
res = a + b
```
Why wasn't this an issue before?

Reply:
Oh interesting, somehow AutogradMeta must be nullptr in that case as well. 
This wasn't an issue before because `init_to_execute` was only called for the dummy node and `roots` never contained invalid edges. It also seems like the rest of the code is robust against invalid edges.

Reply:
Ho right because we would fail earlier in the code if one of the `roots` did not have requires_grad=True.